tdigest.go

```diff
@@ -335,3 +345,96 @@ func (cs *centroidSet) MergeInto(other TDigest) {
 		other.Add(c.mean, c.count)
 	}
 }
+
+func (cs *centroidSet) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	w := &stickyErrWriter{w: buf, err: nil}
+	binary.Write(w, binary.LittleEndian, int32(1))
+
+	var min, max float64
+	if len(cs.centroids) > 0 {
+		min = cs.centroids[0].mean
+		max = cs.centroids[len(cs.centroids)-1].mean
+	}
+
+	binary.Write(w, binary.LittleEndian, min)
+	binary.Write(w, binary.LittleEndian, max)
+	binary.Write(w, binary.LittleEndian, float64(cs.compression))
+	binary.Write(w, binary.LittleEndian, int32(len(cs.centroids)))
+	if w.err != nil {
+		return nil, w.err
+	}
+	for _, c := range cs.centroids {
+		binary.Write(w, binary.LittleEndian, int64(c.count))
```

Comment:
I really like that this is an `int64`. Consider changing the types of `centroidSet.countTotal` and `centroid.Count` to `int64` rather than `int`—though maybe that should happen as a separate maintenance commit.

Reply:
Right, done in this revision. I expect this to be v2.0.0 when it lands anyway.